src/Microsoft.ML/MemoryCollection.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+
+namespace Microsoft.ML
+{
+    public class MemoryCollection
+    {
+        /// <summary>
+        /// Creates memory collection loader.
+        /// </summary>
+        public static MemoryCollectionLoader<T> Create<T>(IList<T> data) where T:class
+        {
+            return new MemoryCollectionLoader<T>(data);
+        }
+
+        /// <summary>
+        /// Creates memory collection loader.
+        /// </summary>
+        public static MemoryCollectionLoader<T> Create<T>(IEnumerable<T> data) where T : class
```

Comment:
>Loader [](start = 38, length = 6)

The word "Loader" is unfortunate. We've always used this too describe things that load from a stream. Loaders also have a data model, this thing won't. You're introducing new capabilities because the original design of the API that assumed that there would always be a data source situated from a stream and loaded through an `IDataLoader` was insufficient -- but this also means that we may want to call this something else. We could rename `ILearningPipelineLoader` to something else... as `ILearningPipelineSource`, perhaps? Then a `ILearningPipelineLoader` would then be a specialization of that source interface. Anyway, loader is not an appropriate name for this broader perspective we're introducing in your PR.

(Though I note that, somehow, that in this new API the `ILearningPipelineLoader` interface does not have a mechanism getting the data model out, that should probably be added... but that should probably be a separate issue.